src/optimizer/rule/PushFilterDownGetNbrsRule.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef OPTIMIZER_RULE_PUSHFILTERDOWNGETNBRSRULE_H_
+#define OPTIMIZER_RULE_PUSHFILTERDOWNGETNBRSRULE_H_
+
+#include <memory>
+
+#include "optimizer/OptRule.h"
+
+namespace nebula {
+namespace graph {
+class GetNeighbors;
+}   // namespace graph
+
+namespace opt {
+
+class PushFilterDownGetNbrsRule final : public OptRule {
+public:
+    static std::unique_ptr<OptRule> kInstance;
+
+    bool match(const OptGroupExpr *groupExpr) const override;
+    Status transform(graph::QueryContext *qctx,
+                     const OptGroupExpr *groupExpr,
+                     TransformResult *result) const override;
```

Comment:
A question, Way `const` of method transform? Doing so will not allow set private variables in class xxxxRule. WDYT?

Reply:
Each rule will be used by many queries, and we register each rule in the singleton way into rule set. so change state of each rule will affect optimization of other queries. we should keep each rule stateless.